chore: npm-publish: replace "npm install --legacy-peer-deps" with "npm ci"

thecopy opened the pull request:
Motivation:
* Self-Service email-service npm package fails if `--legacy-peer-deps` is used
* `npm install` does not install the same versions as package.json in practice: virtually always we use `^#.#.#` in `package.json` - using `npm install` will install the latest minor version while `npm ci` will respect the lock file

thecopy commented:
@zhaow-de comment said some package with without legacy peer deps - is this still the case?

zhaow-de commented:
> @zhaow-de comment said some package with without legacy peer deps - is this still the case?

could u help to check? https://github.com/alloy-ch/eslint-config-alloy/blob/c51556bdd245b771bff206919cb32d4aa002aa56/README.md

thecopy commented:
> > @zhaow-de comment said some package with without legacy peer deps - is this still the case?
> 
> could u help to check? https://github.com/alloy-ch/eslint-config-alloy/blob/c51556bdd245b771bff206919cb32d4aa002aa56/README.md

Seems to work

Locally:
```
~/dev/alloy/eslint-config-alloy main#v0.1.6                                                                                                                     
❯ npm ci

added 210 packages, and audited 211 packages in 1s
pick 4b0053f test: dont merge: test npm ci in ops-ci-shared

74 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

CodeBuild deploy: deploying specific commit which has `npm ci` instead: https://github.com/alloy-ch/eslint-config-alloy/pull/23/commits/8c7974324075a06b516fed7c2615b78b94cb281d:

```
Publishing npm package from .
added 210 packages, and audited 211 packages in 2s
74 packages are looking for funding
  run `npm fund` for details
found 0 vulnerabilities
> @alloy-ch/eslint-config-alloy@0.1.7-alpha build
> echo "OK. No need to run any extra build step."
OK. No need to run any extra build step.
(...)
npm notice Publishing to https://npm.pkg.github.com with tag unstable and default access

+ @alloy-ch/eslint-config-alloy@0.1.7-alpha
```
